src/Controller/ListProductsAction.php

```diff
@@ -30,9 +30,13 @@ public function __invoke(Request $request): Response
     {
         $catalog = $this->getDefaultCatalog();
         $locale = $this->guessCurrentLocaleQuery->guess();
-        $products = $catalog->enabled
-            ? $this->fetchProductsQuery->fetch($locale)
-            : [];
+
+        if ($catalog->enabled) {
+            $productsIdentifiers = $this->catalogApiClient->getProductIdentifiers($catalog->id);
```

Comment:
If I'm not mistaken, the get product identifiers endpoint supports `limit`, no need to ask more than needed and slice the response.